src/sync/events/utils.ts

```diff
@@ -64,11 +73,23 @@ export const fetchBlock = async (chainId: number, blockNumber: number, force = f
                 gasPrice,
                 gasUsed,
                 gasFee,
+                gasLimit,
+                nonce: 1,
               };
             }
           );
+
+          const receipts: TransactionReceipt[] = [];
+
+          if (config.shouldProcessTransactionReceipts) {
+            for await (const tx of block.transactions) {
+              const receipt = await getProvider(chainId).getTransactionReceipt(tx.hash);
```

Comment:
We shouldn't do this. If we want to do this, we don't need `getLogs` method. We can iterate all transactions in the block, which we don't want to do.

Reply:
This will only run conditionally if we set the env variable `PROCESS_TRANSACTION_RECEIPTS` to 1. We would keep this disabled.

But, I've still kept this part so that if we want reliable txn status values, getting transaction receipts is the only way to get it and we would just need to change the env variable's value.

Also, I've removed the line where default value is being set from hardcoded value (1)